frontend/app/routes/$lang+/_protected+/personal-information+/email-address+/edit.tsx

```diff
@@ -0,0 +1,208 @@
+      } else if (val.emailAddress !== val.confirmEmailAddress) {
+        ctx.addIssue({ code: z.ZodIssueCode.custom, message: 'email-match', path: ['confirmEmail'] });
+      }
+    });
+
+  const formData = Object.fromEntries(await request.formData());
+  const expectedCsrfToken = String(session.get('csrfToken'));
+  const submittedCsrfToken = String(formData['_csrf']);
+
+  if (expectedCsrfToken !== submittedCsrfToken) {
+    log.warn('Invalid CSRF token detected; expected: [%s], submitted: [%s]', expectedCsrfToken, submittedCsrfToken);
+    throw new Response('Invalid CSRF token', { status: 400 });
+  }
+
+  const parsedDataResult = formDataSchema.safeParse(formData);
+
+  if (!parsedDataResult.success) {
+    instrumentationService.countHttpStatus('email-address.confirm', 400);
+    return json({
+      errors: parsedDataResult.error.format(),
+      formData: formData as Partial<z.infer<typeof formDataSchema>>,
+    });
+  }
+
+  session.set('newEmailAddress', parsedDataResult.data.emailAddress);
+
+  instrumentationService.countHttpStatus('email-address.confirm', 302);
+
+  const userInfoToken: UserinfoToken = session.get('userInfoToken');
+  const personalInformationRouteHelpers = getPersonalInformationRouteHelpers();
+  const personalInformationService = getPersonalInformationService();
+  const personalInformation = await personalInformationRouteHelpers.getPersonalInformation(userInfoToken, params, request, session);
+
+  const newPersonalInformation = {
+    ...personalInformation,
+    emailAddress: session.get('newEmailAddress'),
+  };
+  await personalInformationService.updatePersonalInformation(userInfoToken.sin ?? '', newPersonalInformation);
+
+  const idToken: IdToken = session.get('idToken');
+  getAuditService().audit('update-data.email-address', { userId: idToken.sub });
+
+  session.unset('newEmailAddress');
+
+  instrumentationService.countHttpStatus('email-address.confirm', 302);
+  return redirectWithSuccess(getPathById('$lang+/_protected+/personal-information+/index', params), 'personal-information:email-address.edit.updated-notification');
+}
+
+export default function EmailAddressEdit() {
+  const { csrfToken, emailAddress } = useLoaderData<typeof loader>();
+  const params = useParams();
+  const fetcher = useFetcher<typeof action>();
+  const errorSummaryId = 'error-summary';
+
+  const { t } = useTranslation(handle.i18nNamespaces);
+
+  const defaultValues = {
+    emailAddress: fetcher.data?.formData.emailAddress ?? emailAddress,
+    confirmEmailAddress: fetcher.data?.formData.confirmEmailAddress ?? emailAddress,
+  };
+
+  /**
+   * Gets an error message based on the provided internationalization (i18n) key.
+   *
+   * @param errorI18nKey - The i18n key for the error message.
+   * @returns The corresponding error message, or undefined if no key is provided.
+   */
+  function getErrorMessage(errorI18nKey?: string): string | undefined {
```

Comment:
The way that error messages are rendered has since been refactored. I suggest we change this component to do the same for consistency.

See https://github.com/DTS-STN/canadian-dental-care-plan/blob/a0c43159f1e10002648c74ededcdd025f0cf6ea4/frontend/app/routes/%24lang%2B/_public%2B/apply%2B/%24id%2B/communication-preference.tsx#L169

Reply:
Fixed in commit https://github.com/DTS-STN/canadian-dental-care-plan/commit/99c426fa45474172e0f4fe2d512598318b8cca7f.